Applications/Spire/Source/Ui/FontBox.cpp

```diff
@@ -117,6 +117,10 @@ const std::shared_ptr<ValueModel<QFont>>& FontBox::get_current() const {
   return m_current;
 }
 
+void FontBox::showEvent(QShowEvent* event) {
+  m_font_family_box->setFocus();
```

Comment:
What is the reason for this? I am concerned that anytime this is hidden and shown it will steal focus from anything in the entire application.

Reply:
It is used to focus on the FontFamilyBox when the FontBox is shown. Since the AdaptiveBox inside the FontBox changes the layout, the tab order gets updated, causing the FontBox to lose the default input focus. You are right. This approach may steal focus from other widgets when it is shown. I'll look into finding a different solution for this.